### PR TITLE
detect: unify ipc config and model

### PR DESCRIPTION
**Pull request by slawblauciak**
Previously the component required two different IPCs
in order to be configured properly, which might be inconvenient
in certain cases.

Now, there's only one IPC that carries both the comp config and
the detection model.

This however removes the ability to set the config during the creation
of the component, since the IPC is sent in chunks.

Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

File: src/audio/detect_test.c [hunk 1]
```diff
@@ -654,7 +545,7 @@ struct comp_driver comp_keyword = {
 	},
 };
 
-static void sys_comp_keyword_init(void)
+UT_STATIC void sys_comp_keyword_init(void)
```
Comment (jajanusz) [inline src/audio/detect_test.c]: Looks like this + changes in src/include/sof/audio/detect_test.h are fix for making symbol accessible from tests. Can it be done in separate commit? First fix then refactor?

File: src/audio/detect_test.c [hunk 2]
```diff
@@ -430,74 +370,26 @@ static int test_keyword_get_config(struct comp_dev *dev,
...
-			trace_keyword_error("test_keyword_get_model() error: "
-					    "invalid size %d", bs);
-			return -EINVAL;
-		}
-
-		crc = crc32(cd->load_memory, cd->config.load_memory_size);
-
-		trace_keyword("test_keyword_get_model() crc: 0x%X", crc);
+	/* Copy back to user space, without the model */
+	bs = sizeof(struct sof_detect_test_config);
```
Comment (keyonjie) [inline src/audio/detect_test.c]: bs = cd->config->size; ?

Comment (slawblauciak) [reply src/audio/detect_test.c]: size contains the model

Comment (keyonjie) [reply src/audio/detect_test.c]: @slawblauciak we need to get this data(model) and send back to host driver side to dumping to end  user, so the size should be the whole size.

Comment (q009) [reply src/audio/detect_test.c]: The mailbox is not large enough for the entire model, it can't be copied there. And I can't exactly split it into chunks.
